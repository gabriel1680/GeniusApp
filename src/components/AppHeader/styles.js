import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    container: {
        position: 'absolute',
        top: 10,
        justifyContent: 'center',
        alignItems: 'center'
    },
    title: {
        textAlign: 'center',
        fontSize: 40,
        fontWeight: 'bold',
        color: '#000',
        marginTop: 15
    },
    subTitle: {
        textAlign: 'center',
        fontSize: 20,
        fontWeight: 'bold',
        color: '#000',
        marginTop: 15
    }
});

export default styles;