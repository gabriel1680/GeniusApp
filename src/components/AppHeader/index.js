import React from 'react';
import { Text, View } from 'react-native';

import styles from './styles';

const AppHeader = () => {
    return (
        <View style={styles.container}>
            <Text style={styles.title}>GeniusApp</Text>
            <Text style={styles.subTitle}>Teste Sua Mem√≥ria!</Text>
        </View>
    );
};

export default AppHeader;