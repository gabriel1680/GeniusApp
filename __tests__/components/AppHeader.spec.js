/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import AppHeader from '../../src/components/AppHeader';
import { render } from '@testing-library/react-native';

it('renders correctly', () => {
  renderer.create(<AppHeader />);
});

it('should have the title and subtitle', () => {
  const { getByText } = render(<AppHeader />);
  getByText('GeniusApp');
  getByText('Teste Sua MemÃ³ria!');
});

it('should match snapshot', () => {
  const header = render(<AppHeader />);
  expect(header).toMatchSnapshot();
});
