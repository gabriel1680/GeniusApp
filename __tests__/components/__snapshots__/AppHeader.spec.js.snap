// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader (unit) should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "position": "absolute",
      "top": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 40,
        "fontWeight": "bold",
        "marginTop": 15,
        "textAlign": "center",
      }
    }
  >
    GeniusApp
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": 15,
        "textAlign": "center",
      }
    }
  >
    Teste Sua Memória!
  </Text>
</View>
`;

exports[`should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "position": "absolute",
      "top": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 40,
        "fontWeight": "bold",
        "marginTop": 15,
        "textAlign": "center",
      }
    }
  >
    GeniusApp
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": 15,
        "textAlign": "center",
      }
    }
  >
    Teste Sua Memória!
  </Text>
</View>
`;
